### ISSUE TITLE
Ability to call after route.pre

### ISSUE THREAD

**Issue by petermilan** [mark-bradshaw/mrhorse#33]
Hello there, I think it might be usefull to have some applyPoint which would be called beforeHandler but after *route.pre*  , not sure if it is doable, but it would be handy to have available what is inside of *request.pre*

**Comment by mark-bradshaw** [comment]
I'm not sure I'm understanding what you mean, exactly, but you can apply a policy at any part of the request/response life cycle.  The applyPoint can be any of the events that Hapi supports.

**Comment by petermilan** [comment]
I know that part works perfect, but in hapy you can define code which should happen before handler in 2 ways.
1) is the request/response lifecycle - which works
2) is to define pre handler inside of the route definition https://hapijs.com/api#route-prerequisites like this:

```
const Hapi = require('hapi');
const server = new Hapi.Server();
server.connection({ port: 80 });

const pre1 = function (request, reply) {

    return reply('Hello');
};

const pre2 = function (request, reply) {

    return reply('World');
};

const pre3 = function (request, reply) {

    return reply(request.pre.m1 + ' ' + request.pre.m2);
};

server.route({
    method: 'GET',
    path: '/',
    config: {
        pre: [
            [
                // m1 and m2 executed in parallel
                { method: pre1, assign: 'm1' },
                { method: pre2, assign: 'm2' }
            ],
            { method: pre3, assign: 'm3' },
        ],
        handler: function (request, reply) {

            return reply(request.pre.m3 + '\n');
        }
    }
});
```

and as you  can see on this diagram http://freecontent.manning.com/hapi-js-in-action-diagram/
the order is:
1 onPreHandler()
2 config.pre
3 handler code

it would be nice to have ability to defie in mrhorse a policy between config.pre and the handler code.

**Comment by mark-bradshaw** [comment]
Ah I see what you mean.  There's no way to do that.  Hapi fires off prehandlers right before the handler, and there's no life cycle event that occurs between as far as I know.  Basically, if you want to use MrHorse, don't use Hapi's pre handlers.

**Comment by petermilan** [comment]
there could be perhaps a chance to do this, but mhorse would need to modify route, and push its policy to the `route.pre` array (as the last item), but I understand that it is sort of quite hacking around. Just left here this idea.
